InnerEye/ML/reports/classification_report.py

```diff
@@ -52,84 +54,220 @@ class ReportedMetrics(Enum):
+    """
+    if ax is None:
+        ax = plt.gca()
+    x_grid = np.linspace(0, 1, 101)
+    interp_ys = []
+    line_handles = []
+    for index, labels_and_model_outputs in enumerate(all_labels_and_model_outputs):
+        x_values, y_values = scoring_fn(labels_and_model_outputs)
+        interp_ys.append(np.interp(x_grid, x_values, y_values))
+        handle, = ax.plot(x_values, y_values, lw=1)
+        line_handles.append(handle)
+
+    confidence_interval_quantiles = [.5 - confidence_interval_width / 2, .5, .5 + confidence_interval_width / 2]
+    y_lo, y_mid, y_hi = quantile(interp_ys, confidence_interval_quantiles, axis=0)
+    h1 = ax.fill_between(x_grid, y_lo, y_hi, color='k', alpha=.2, lw=0)
+    h2, = ax.plot(x_grid, y_mid, 'k', lw=2)
+    summary_handle = (h1, h2)
+    return line_handles, summary_handle
+
+
+def plot_pr_and_roc_curves_crossval(all_labels_and_model_outputs: Sequence[LabelsAndPredictions],
+                                    axs: Optional[Sequence[Axes]] = None) -> None:
+    """
+    Given a list of LabelsAndPredictions objects, plot the corresponding ROC and PR curves, along with median line and
+    shaded 80% confidence interval (computed over TPRs and precisions for each fixed FPR and recall value).
+    :param all_labels_and_model_outputs: Collection of ground-truth labels and model predictions (e.g. for various
+    cross-validation runs).
+    :param axs: Pair of axes objects onto which to plot the ROC and PR curves, respectively. New axes are created by
+    default.
+    """
+    if axs is None:
+        _, axs = plt.subplots(1, 2)
+
+    def get_roc_xy(labels_and_model_outputs: LabelsAndPredictions) -> Tuple[np.ndarray, np.ndarray]:
+        fpr, tpr, thresholds = roc_curve(labels_and_model_outputs.labels, labels_and_model_outputs.model_outputs)
+        return fpr, tpr
+
+    def get_pr_xy(labels_and_model_outputs: LabelsAndPredictions) -> Tuple[np.ndarray, np.ndarray]:
+        precision, recall, thresholds = precision_recall_curve(labels_and_model_outputs.labels,
+                                                               labels_and_model_outputs.model_outputs)
+        return recall[::-1], precision[::-1]  # inverted to be in ascending order
+
+    confidence_interval_width = .8
+    line_handles, summary_handle = plot_scores_and_summary(all_labels_and_model_outputs,
+                                                           scoring_fn=get_roc_xy, ax=axs[0],
+                                                           confidence_interval_width=confidence_interval_width)
+    plot_scores_and_summary(all_labels_and_model_outputs, scoring_fn=get_pr_xy, ax=axs[1],
+                            confidence_interval_width=confidence_interval_width)
+
+    line_labels = [f"Split {split_index}" for split_index in range(len(all_labels_and_model_outputs))]
+    axs[0].legend(line_handles + [summary_handle],
+                  line_labels + [f"Median, {100 * confidence_interval_width:g}% CI"])
```

Comment:
Same comment, from a statistical point of view it's not a CI. Alternatively you could say you plot the interquartile range as shaded area

Reply:
(see replies above)